packages/typography/src/Link.tsx

```diff
@@ -119,12 +119,14 @@ const Link: ExtendableBox<LinkProps, 'a'> = ({
     );
   }
 
+  const elementProps = href
+    ? ({ as: 'a', href, target } as const)
```

Comment:
Is the `as const` what was necessary for typing due to the changes to Box?

Reply:
No, this fix is independent of the changes to Box. TypeScript sees the previous code as passing in `{as: string, href: string | undefined, target: string | undefined}`, which does not satisfy any of the overload variants. The fix is typed as `{as: 'a', href: string, targe: string} | {as: 'span'}` and since each type in the union satisfies an overload variant, the whole type does.